src/query/service/src/interpreters/interpreter_txn_commit.rs

```diff
@@ -50,3 +61,107 @@ impl Interpreter for CommitInterpreter {
+    if is_active {
+        let catalog = ctx.get_default_catalog()?;
+
+        let req = ctx.txn_mgr().lock().req();
+
+        let update_summary = {
+            let table_descriptions = req
+                .update_table_metas
+                .iter()
+                .map(|(req, _)| (req.table_id, req.seq, req.new_table_meta.engine.clone()))
+                .collect::<Vec<_>>();
+            let stream_descriptions = req
+                .update_stream_metas
+                .iter()
+                .map(|s| (s.stream_id, s.seq, "stream"))
+                .collect::<Vec<_>>();
+            (table_descriptions, stream_descriptions)
+        };
+
+        let mismatched_tids = {
+            ctx.txn_mgr().lock().set_auto_commit();
+            let ret = catalog.retryable_update_multi_table_meta(req).await;
+            if let Err(ref e) = ret {
+                // other errors may occur, especially the version mismatch of streams,
+                // let's log it here for the convenience of diagnostics
+                error!(
+                    "Non-recoverable fault occurred during table metadata update: {}",
+                    e
+                );
+            }
+            ret?
+        };
+
+        match &mismatched_tids {
+            Ok(_) => {
+                info!(
+                    "Transaction committed successfully, updated targets: {:?}",
+                    update_summary
+                );
+            }
+            Err(e) => {
+                let err_msg = format!(
+                    "Due to concurrent transactions, explicit transaction commit failed. Conflicting table IDs: {:?}",
+                    e.iter().map(|(tid, _, _)| tid).collect::<Vec<_>>()
+                );
+                info!(
+                    "Transaction commit failed due to concurrent modifications. Conflicting table IDs: {:?}",
+                    e
+                );
+                return Err(ErrorCode::TableVersionMismatched(format!("{}", err_msg)));
+            }
+        }
+        let need_purge_files = ctx.txn_mgr().lock().need_purge_files();
+        for (stage_info, files) in need_purge_files {
+            try_purge_files(ctx.clone(), &stage_info, &files).await;
+        }
+    }
+    Ok(())
+}
+
+#[async_backtrace::framed]
+async fn try_purge_files(ctx: Arc<dyn TableContext>, stage_info: &StageInfo, files: &[String]) {
+    let start = Instant::now();
+    let op = init_stage_operator(stage_info);
+
+    match op {
+        Ok(op) => {
+            let file_op = Files::create(ctx, op);
+            if let Err(e) = file_op.remove_file_in_batch(files).await {
+                error!("Failed to delete files: {:?}, error: {}", files, e);
+            }
+        }
+        Err(e) => {
+            error!("Failed to initialize stage operator, error: {}", e);
+        }
+    }
+
+    let elapsed = start.elapsed();
+    info!(
+        "[SQL-BINDER] Purged {} files, operation took {:?}",
```

Comment:
After the moving, this log prefix is not `[SQL-BINDER]`, I suggest to remove here:
```suggestion
        "Purged {} files, operation took {:?}",
```